config/config_test.go

```diff
@@ -0,0 +1,235 @@
+		got, err := interpolate(testLookupEnv, "${GORAY}${THREADS_COUNT}")
+
+		assert.Equal(t, "some_value5", got)
+		assert.Nil(t, err)
+	})
+
+	t.Run("valid: two env variables found within the same string", func(t *testing.T) {
+		got, err := interpolate(testLookupEnv, "${GORAY}${THREADS_COUNT}")
+
+		assert.Equal(t, "some_value5", got)
+		assert.Nil(t, err)
+	})
+
+	t.Run("valid: no env variables within the string", func(t *testing.T) {
+		got, err := interpolate(testLookupEnv, "${NOT AN ENV")
+
+		assert.Equal(t, "${NOT AN ENV", got)
+		assert.Nil(t, err)
+	})
+
+	t.Run("invalid: env variable not found", func(t *testing.T) {
+		got, err := interpolate(testLookupEnv, "${SOME_NAME}")
+
+		assert.Equal(t, "", got)
+		assert.Equal(t, "environment variables not found: SOME_NAME", err.Error())
+	})
+
+	t.Run("invalid: one env variable found, one not found", func(t *testing.T) {
+		got, err := interpolate(testLookupEnv, "${GORAY}${SOME_NAME}")
+
+		assert.Equal(t, "", got)
+		assert.Equal(t, "environment variables not found: SOME_NAME", err.Error())
+	})
+
+	t.Run("invalid: multiple env variables not found", func(t *testing.T) {
+		got, err := interpolate(testLookupEnv, "${SOME_NAME1} ${SOME_NAME2} ${SOME_NAME3}")
+
+		assert.Equal(t, "", got)
+		assert.Equal(t, "environment variables not found: SOME_NAME1, SOME_NAME2, SOME_NAME3", err.Error())
+	})
+}
+
+func withTestReadFile(rc *ReadConf) {
+	rc.readFile = testReadFile
+}
+
+func testReadFile(name string) ([]byte, error) {
+	return []byte(`
+		[workspace]
+		version = "1.0.193"
+		root= "."
+		policies="./policy"
+		ignore-list=[".git", ".docker"]
+		go-arch="HOME: ${GORAY}"
+		
+		[analysis]
+		KRAIL1001 = {level = "info", target= "./${GORAY}"}
+		
+		[analysis.SE]
+		level = "critical"
+		target = "./internal"
+	`), nil
+}
+
+func withTestReadFileErr(rc *ReadConf) {
+	rc.readFile = testReadFileErr
+}
+
+func testReadFileErr(name string) ([]byte, error) {
+	return nil, errors.New("read file error")
+}
+
+func withTestReadToml(rc *ReadConf) {
+	rc.readToml = testReadToml
+}
+
+func testReadToml(data string, v interface{}) (toml.MetaData, error) {
+	return toml.MetaData{}, nil
+}
+
+func withTestReadTomlErr(rc *ReadConf) {
+	rc.readToml = testReadTomlErr
+}
+
+func testReadTomlErr(data string, v interface{}) (toml.MetaData, error) {
+	return toml.MetaData{}, errors.New("read toml error")
+
+}
+
+func withTestInterpolate(rc *ReadConf) {
+	rc.interpolate = testInterpolate
+}
+
+func testInterpolate(lookup LookupFn, strWithEnvs string) (string, error) {
+	return "interpolated string", nil
+}
+
+func withTestInterpolateErr(rc *ReadConf) {
+	rc.interpolate = testInterpolateErr
+}
+
+func testInterpolateErr(lookup LookupFn, strWithEnvs string) (string, error) {
+	return "", errors.New("interpolate error")
+}
+
+func withTestLookup(rc *ReadConf) {
+	rc.lookup = testLookup
+}
+
+func testLookup(key string) (string, bool) {
+	return "", false
+}
```

Comment:
this is a mock function, so the name for all mock should be with a `mock* `prefix and in the top of the file after the constants 

smth like mockLookup